### PR TITLE
Fixes #6367: Handle relay server in rudder-tests

### PR DESCRIPTION
**Pull request by VinceMacBuche**
http://www.rudder-project.org/redmine/issues/6367

### CODE REVIEW

File: rtf [hunk 1]
```diff
@@ -51,20 +51,27 @@ def host_lines(platform, hosts, pf_id):
   pf = platform.split('.')[0]
   lines=[]
   i=1
-  for hostname, host_obj in hosts.iteritems():
+  host_list = sorted(hosts.keys(), reverse=True)
+  hostnames = " ".join(host_list)
+  for hostname in host_list:
+    host_obj = hosts[hostname]
     host = host_obj.info
```
Comment (peckpeck) [inline rtf]: Thos 2 lines could be merged into host = hosts[hostname].info

File: rtf [hunk 2]
```diff
@@ -137,29 +144,29 @@ class Vagrant(Host):
...
+       	  updated = True
+       	elif not updated and re.match(r'### AUTOGEN TAG', line):
+       	  lines.append('### AUTOGENERATED FOR PLATFORM ' + name + " (" + str(max_pf_id+1) + ")\n")
+       	  lines.extend(host_lines(name, hosts, max_pf_id+1))
+       	  lines.append('### END OF AUTOGENERATION FOR ' + name + "\n")
+       	  lines.append("\n### AUTOGEN TAG\n")
+       	  updated = True
+       	else:
+       	  lines.append(line)
+       	line=fd.readline()
```
Comment (peckpeck) [inline rtf]: This is a reindent error, it shouldn't be there

File: rtf [hunk 3]
```diff
@@ -183,7 +190,7 @@ class Platform:
     default = platform_info['default']
     for hostname, host in platform_info.items():
       if hostname == "default":
-          continue
+        continue
```
Comment (peckpeck) [inline rtf]: This too

File: vagrant.rb [hunk 4]
```diff
@@ -39,35 +39,32 @@
...
-  command  = "/vagrant/scripts/cleanbox " + net + "\n"
-  if host_id == 0 then
+  command  = "/vagrant/scripts/cleanbox " + net + ' "' + host_list + "\"\n"
+  if setup == "server" then
     command += "export ALLOWEDNETWORK=" + net + ".0/24\n"
     command += "/vagrant/scripts/rudder-setup setup-server " + version + "\n"
     command += "/vagrant/scripts/create-token\n"
   else
-    command += "/vagrant/scripts/rudder-setup setup-agent " + version + "\n"
+    command += "/vagrant/scripts/rudder-setup setup-"+setup+" " + version + " '" + server + "'\n"
```
Comment (peckpeck) [inline vagrant.rb]: Since we use positional parameters version should be surrounded by ''

File: rtf [hunk 5]
```diff
@@ -319,6 +348,7 @@ def list_scenarii():
 
 if __name__ == "__main__":
   args = docopt.docopt(__doc__)
+  client_path = "rudder-api-client"
```
Comment (peckpeck) [inline rtf]: You should put the "if not os.path.exists(client_path):" block here too

File: rtf [hunk 6]
```diff
@@ -191,12 +198,34 @@ class Platform:
...
+        print(host.run("/var/rudder/cfengine-community/bin/cf-agent -K"))
+        # Environment setup
+        rudder_url = host.get_url()
+        token = host.run('cat /root/rudder-token')
+        setenv(client_path, rudder_url, token)
+        for relay in relay_list:
+          # Accept the relay
+          rcli = "rudder-cli --skip-verify --url=" + rudder_url + " --token=" + token
+          command = rcli + " nodes accept "+ relay
+          check_output(command, shell=True)
```
Comment (peckpeck) [inline rtf]: You should surround check_output with print()

File: rtf (hunk 2 above)
Comment (VinceMacBuche) [inline rtf]: I thought we decided to fix reindent issue here :(

File: rtf (hunk 3 above)
Comment (VinceMacBuche) [inline rtf]: same :(

**Comment by VinceMacBuche** [comment]
Updated

File: rtf (hunk 3 above)
Comment (VinceMacBuche) [inline rtf]: This one is correct :)

**Comment by peckpeck** [comment]
All right !

**Pull request merged**